drivers/modem/alt1250/alt1250.c

```diff
@@ -0,0 +1,1279 @@
+}
+
+/****************************************************************************
+ * Name: alt1250_ioctl
+ ****************************************************************************/
+
+static int alt1250_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
+{
+  FAR struct inode *inode;
+  FAR struct alt1250_dev_s *dev;
+  int ret = OK;
+
+  /* Get our private data structure */
+
+  DEBUGASSERT(filep != NULL && filep->f_inode != NULL);
+  inode = filep->f_inode;
+
+  dev = (FAR struct alt1250_dev_s *)inode->i_private;
+  DEBUGASSERT(dev);
+
+  switch (cmd)
+    {
+      case ALT1250_IOC_POWER:
+        {
+          FAR struct alt_power_s *req = (FAR struct alt_power_s *)arg;
+
+          /* Performs power control or power consumption control
+           * of the modem.
+           */
+
+          ret = alt1250_power_control(dev, req);
+        }
+        break;
+
+      case ALT1250_IOC_SEND:
+        {
+          FAR alt_container_t *req = (FAR alt_container_t *)arg;
+
+          ret = make_altcomcmd_and_send(dev, req);
+        }
+        break;
+
+      case ALT1250_IOC_SETEVTBUFF:
+        {
+          FAR struct alt_evtbuffer_s *buff =
+            (FAR struct alt_evtbuffer_s *)arg;
+          add_evtbuff(dev, buff);
+        }
+        break;
+
+      case ALT1250_IOC_EXCHGCONTAINER:
+        {
+          FAR alt_container_t **container = (FAR alt_container_t **)arg;
+
+          ret = exchange_selectcontainer(dev, container);
+        }
+        break;
+
+      default:
+        ret = -EINVAL;
+        break;
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: alt1250_poll
+ ****************************************************************************/
+
+static int alt1250_poll(FAR struct file *filep, FAR struct pollfd *fds,
+  bool setup)
+{
+  FAR struct inode *inode;
+  FAR struct alt1250_dev_s *dev;
+
+  /* Get our private data structure */
+
+  DEBUGASSERT(filep != NULL && filep->f_inode != NULL);
+  inode = filep->f_inode;
+
+  dev = (FAR struct alt1250_dev_s *)inode->i_private;
+  DEBUGASSERT(dev);
+
+  if (setup)
+    {
+       poll_notify(&fds, 1, POLLIN);
+    }
+
+  return OK;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+FAR void *alt1250_register(FAR const char *devpath,
+  FAR struct spi_dev_s *dev, FAR const struct alt1250_lower_s *lower)
+{
+  FAR struct alt1250_dev_s *priv;
+  int ret;
+
+  priv = (FAR struct alt1250_dev_s *)
+    kmm_malloc(sizeof(struct alt1250_dev_s));
+  if (!priv)
+    {
+      m_err("Failed to allocate instance.\n");
+      return NULL;
+    }
+
+  memset(priv, 0, sizeof(struct alt1250_dev_s));
+
+  priv->spi = dev;
+  priv->lower = lower;
+
+  nxsem_init(&priv->refslock, 0, 1);
+
+  ret = register_driver(devpath, &g_alt1250fops, 0666, priv);
```

Comment:
why not expose the functionality through usrsock? so, user could:

1. use the standard socket api send/recv the data
2. control the specific function through ioctl or setsockopt

Reply:
that is under investigation for next refactor

Reply:
Ok.